### PR TITLE
Fix Colors

### PR DESCRIPTION
**Pull request by mgazza**
Fix various issues with colour mapping. Add some tests based on https://github.com/github/putty/blob/master/test/colours.txt

### CODE REVIEW

File: cmd/fyneterm/theme.go [hunk 1]
```diff
@@ -26,6 +26,10 @@ func (t *termTheme) Color(n fyne.ThemeColorName, v fyne.ThemeVariant) color.Colo
 			return color.NRGBA{R: 0xaa, G: 0xaa, B: 0xaa, A: 0xf6}
 		}
 		return color.NRGBA{R: 0x0a, G: 0x0a, B: 0x0a, A: 0xf6}
+	case theme.ColorNameForeground:
+		// aligns with terminal.basicColors[7]
+		return color.NRGBA{R: 170, G: 170, B: 170, A: 0xff}
```
Comment (andydotxyz) [inline cmd/fyneterm/theme.go:31]: This must return a better colour for the light theme (see the if above).
Run with FYNE_THEME=light and you won't be able to read text any more.

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Thanks for doing this, but I think it has forgotten about light mode.
One item noted is definitely missing code - but I wonder if color.go:94 is dark mode specific too?

File: cmd/fyneterm/theme.go (hunk 1 above)
Comment (mgazza) [reply cmd/fyneterm/theme.go:31]: Thank you I'll take a look

Comment (mgazza) [reply cmd/fyneterm/theme.go]: OK pushed a fix for this

File: cmd/fyneterm/theme.go [hunk 2]
```diff
@@ -23,9 +23,15 @@ func (t *termTheme) Color(n fyne.ThemeColorName, v fyne.ThemeVariant) color.Colo
 			return c
 		}
 		if v == theme.VariantLight {
-			return color.NRGBA{R: 0xaa, G: 0xaa, B: 0xaa, A: 0xf6}
+			return color.NRGBA{R: 85, G: 85, B: 255, A: 0xf6}
```
Comment (andydotxyz) [inline cmd/fyneterm/theme.go:26]: This cannot be right? it replaces what was a light grey with a vivid blue color...

**Review by andydotxyz — COMMENTED** [review]
Legibility is back, thanks - but the bright blue background doesn't seem right

Screenshot 2023-09-13 at 11 54 42

File: cmd/fyneterm/theme.go (hunk 2 above)
Comment (mgazza) [reply cmd/fyneterm/theme.go:26]: I picked a color appropriate for a light-mode terminal. I felt it was subjective. maybe we should put all this into a config file?

Comment (andydotxyz) [reply cmd/fyneterm/theme.go:26]: What was wrong with the light grey background color? in light node most apps have a nearly-white background with dark text.

Comment (mgazza) [reply cmd/fyneterm/theme.go]: ok this has been resolved

**Comment by Jacalz** [comment]
FYI @mgazza. We try to avoid force pushing unless we have merge conflicts or really need to do so for some specific reason. GitHub has a tendency to loose track of review comments sometimes when you force push. If it is about squashing commits we can do so when we merge if necessary :)

**Comment by mgazza** [comment]
I rebased and fixed a conflict. It only looses comments if the diff can't track.

**Comment by andydotxyz** [comment]
I think we discussed before, it doesn't make sense to be using the dark theme to look up a colour for the light theme.
Just getting `theme.ForegroundColor()` should get a suitable foreground colour for the current theme.

**Comment by mgazza** [comment]
> I think we discussed before, it doesn't make sense to be using the dark theme to look up a colour for the light theme. Just getting `theme.ForegroundColor()` should get a suitable foreground colour for the current theme.

Apologies, I thought I had reverted that bit. Fixup inbound.

**Comment by andydotxyz** [comment]
With that reverted you are drawing white text on a light grey background in light mode... 

Look at the Color method - see how it handles the background colour for light and dark mode but then drops down to a "force dark mode" fallback...

All you need is to not force the colour variant for foreground text, not sure where the confusion is here.

**Comment by andydotxyz** [comment]
> Right, but that's how it is currently in master. I have not addressed/changed that at all. Its not a top priority for me right now, so I may address this in another PR

My apologies, it must have been broken at some point in the past - I think the GitHub force pushes were showing me diffs that were not actually correct for the PR

**Review by andydotxyz — APPROVED** [review]

**Comment by andydotxyz** [comment]
> so I may address this in another PR

I've fixed it on master already - not sure how the light mode got broken in the past.

**Pull request merged**